ollama_dl.py

```diff
@@ -40,43 +39,82 @@ class DownloadJob:
     blob_url: str
     size: int
 
+async def download_with_resume(client: httpx.AsyncClient, url: str, dest_path: pathlib.Path, total_size: int, progress, task):
+    temp_path = dest_path.with_suffix(f".tmp-{time.time()}")
+    headers = {}
+    mode = "wb"
+    
+    if temp_path.exists():
+        current_size = temp_path.stat().st_size
+        if current_size < total_size:
+            headers["Range"] = f"bytes={current_size}-"
+            mode = "ab"
+            progress.update(task, completed=current_size)
+        else:
+            temp_path.unlink()
```

Comment:
I'm not sure why we'd want to remove the tempfile and start anew if it already looks complete?